### ISSUE TITLE
Use composite run steps action to reuse steps common to each workflow

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#1528]
https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action#creating-an-action-metadata-file

It'd be nice if we could reduce the amount of duplicated logic we have across each of the various CI workflow files. We should be able to split out most of the common parts into reusable actions.

**Comment by ppalaga** [comment]
Is my understanding correct that each of those composite actions needs a separate git repo?

**Comment by jamesnetherton** [comment]
> Is my understanding correct that each of those composite actions needs a separate git repo?

No, you can add them to the same repo with each action in their own directory and reference them. E.g:

```
    steps:
      - name: Clean VM
        uses: ./.github/actions/cleanup-vm
```

**Comment by ppalaga** [comment]
OK, thanks, that makes them really handy!
